test/unit/test_disco_acm.py

```diff
@@ -8,33 +8,63 @@
 from disco_aws_automation.disco_acm import (
     CERT_SUMMARY_LIST_KEY,
     CERT_ARN_KEY,
-    DOMAIN_NAME_KEY
+    DOMAIN_NAME_KEY,
+    CERT_ALT_NAMES_KEY,
+    CERT_KEY
 )
 
 TEST_DOMAIN_NAME = 'test.example.com'
+TEST_ALT_DOMAIN_NAME = 'test2.example.com'
 TEST_DEEP_DOMAIN_NAME = 'a.deeper.test.example.com'
 TEST_WILDCARD_DOMAIN_NAME = '*.example.com'
-TEST_CERTIFICATE_ARN_ACM_EXACT = "arn:aws:acm::123:exact"
-TEST_CERTIFICATE_ARN_ACM_WILDCARD = "arn:aws:acm::123:wildcard"
+TEST_CERTIFICATE_ARN_ACM_EXACT = 'arn:aws:acm::123:exact'
+TEST_CERTIFICATE_ARN_ACM_WILDCARD = 'arn:aws:acm::123:wildcard'
+TEST_MULTI_CERT_ARN_ACM = 'arn:aws:acm::123:multi'
+TEST_MULTI_DOMAIN_NAME = 'multi.foo.com'
+TEST_MULTI_ALT_DOMAIN_NAME = 'foo.com'
 
 TEST_CERT = {CERT_ARN_KEY: TEST_CERTIFICATE_ARN_ACM_EXACT, DOMAIN_NAME_KEY: TEST_DOMAIN_NAME}
 TEST_WILDCARD_CERT = {CERT_ARN_KEY: TEST_CERTIFICATE_ARN_ACM_WILDCARD,
                       DOMAIN_NAME_KEY: TEST_WILDCARD_DOMAIN_NAME}
+TEST_MULTI_CERT = {CERT_ARN_KEY: TEST_MULTI_CERT_ARN_ACM, DOMAIN_NAME_KEY: TEST_MULTI_DOMAIN_NAME}
 
 
 class DiscoACMTests(TestCase):
-    '''Test disco_acm.py'''
+    """Test disco_acm.py"""
 
     def setUp(self):
         self._acm = MagicMock()
         self.disco_acm = DiscoACM(self._acm)
 
+        self._acm.list_certificates.return_value = {
+            CERT_SUMMARY_LIST_KEY: [TEST_CERT, TEST_WILDCARD_CERT, TEST_MULTI_CERT]
+        }
+
+        def _describe_cert(CertificateArn):
+            cert_data = {
+                TEST_CERTIFICATE_ARN_ACM_WILDCARD: {
+                    CERT_ARN_KEY: TEST_CERTIFICATE_ARN_ACM_WILDCARD,
+                    CERT_ALT_NAMES_KEY: [TEST_WILDCARD_DOMAIN_NAME]
+                },
+                TEST_CERTIFICATE_ARN_ACM_EXACT: {
+                    CERT_ARN_KEY: TEST_CERTIFICATE_ARN_ACM_EXACT,
+                    CERT_ALT_NAMES_KEY: [TEST_DOMAIN_NAME]
+                },
+                TEST_MULTI_CERT_ARN_ACM: {
+                    CERT_ARN_KEY: TEST_MULTI_CERT_ARN_ACM,
+                    CERT_ALT_NAMES_KEY: [TEST_MULTI_DOMAIN_NAME, TEST_MULTI_ALT_DOMAIN_NAME]
+                }
+            }
+
+            return {CERT_KEY: cert_data.get(CertificateArn)}
+
+        self._acm.describe_certificate.side_effect = _describe_cert
```

Comment:
I'm not 100% sure of everything that's going on, but in this case it seems like we could just have

    class MockAcm(object):
        CERT_DATA = { ... all the stuff in ll45-56 ... }
        def list_certificates(self):
            return {CERT_SUMMARY_LIST_KEY: [TEST_CERT, TEST_WILDCARD_CERT, TEST_MULTI_CERT]}
      
        def describe_certificate(self, CertificateArn):
           return {CERT_KEY: self.CERT_DATA.get(CertificateArn)}

then 

     def setUp(self):
          self._acm = MockAcm()

MagicMock gives you all sorts of rope to play with, but if we're not actually using it, just mocking things the old fashioned way seems like it ought to do the job. :-)